src/common/src/buffer/bitmap.rs

```diff
@@ -191,12 +207,17 @@ pub struct Bitmap {
 
     /// Bits are stored in a compact form.
     /// They are packed into `usize`s.
-    bits: Box<[usize]>,
+    ///
+    /// Optimization: If all bits are set to 0 or 1, this field is `None`.
+    bits: Option<Box<[usize]>>,
```

Comment:
I think we can do further optimization, we can just use `count_ones == num_bits` OR `count_ones  == 0` to check if for `all_ones` and `all_zeros` case?
Then it seems we can avoid branching?

Not needed for now I guess. Just my own thoughts if we need to optimize in the future.

Reply:
Do you mean that we first check `count_ones` and then `bits.unwrap_unchecked()`?